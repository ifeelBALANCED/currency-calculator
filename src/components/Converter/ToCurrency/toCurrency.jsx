import React from 'react';
import {Select} from "@chakra-ui/react";
const ToCurrency = ({onSelectChange, toCurrency, allCurrencies}) => {
    return (
        <Select
            name="to"
            onChange={onSelectChange}
            value={toCurrency}
        >
            {allCurrencies}
        </Select>
    );
};

export default ToCurrency;
