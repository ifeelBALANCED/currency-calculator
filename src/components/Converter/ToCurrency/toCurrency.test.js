import React from "react";
import ToCurrency from "./toCurrency";

// initialize test props
const props = {
    value: 0,
    onChange: () => {
    },
}
// initialize shallow copy of ToCurrency component
const setUp = (props) => shallow(<ToCurrency {...props}/>)
describe('should render FromCurrency (Select) component', () => {
    // checks if component has props
    describe('Has props', () => {
        let component = setUp(props)
        it('should render select element', () => {
            const select = component.find('Select')
            expect(select).toHaveLength(1)
        });
        // checks if component rendered with props
        it('should render FromCurrency component with props', () => {
            component = shallow(<ToCurrency name="test" onChange={() => {
            }} value={0}/>)
            expect(component).toMatchSnapshot()
        });
        // check if onSelectChange function is called with a value
        it('should call onSelectChange function with a value', () => {
            const evt = {
                target: {value: "MXN"}
            }
            const mockFn = jest.fn();
            const component = mount(<ToCurrency name="to" onSelectChange={mockFn}/>);
            component.find('select').simulate('change', evt)
            expect(mockFn.mock.calls.length).toBe(1)
        });
    });
});
