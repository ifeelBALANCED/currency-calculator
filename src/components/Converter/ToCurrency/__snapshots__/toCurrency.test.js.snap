// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FromCurrency (Select) component Has props should call onSelectChange function with a value 1`] = `ShallowWrapper {}`;

exports[`should render FromCurrency (Select) component Has props should render FromCurrency component with props 1`] = `
<Select
  name="to"
/>
`;
